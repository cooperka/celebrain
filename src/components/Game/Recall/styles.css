.Recall img {
  height: 30em;
}

.Recall .subtitle {
  margin-bottom: 0;
}
