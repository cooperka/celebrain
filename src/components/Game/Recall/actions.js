// --- Action types

import { setGameState } from '../actions';
import { gameState } from '../reducers';

export const actionTypes = {
  RECALL_NEXT: 'RECALL_NEXT',
};

// --- Action creators

export function next() {
  return (dispatch, getState) => {
    const { recall: { currIndex, numItems } } = getState();
    const nextIndex = currIndex + 1;

    if (nextIndex >= numItems) {
      dispatch(setGameState(gameState.RESULTS));
    } else {
      dispatch({ type: actionTypes.RECALL_NEXT });
    }
  };
}
