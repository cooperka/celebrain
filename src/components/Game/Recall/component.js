// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { TextField } from 'material-ui';

import { next, changedInputText } from './actions';

import './styles.css';

type Props = {
  imageData: any,

  imageOrder: any,
  currIndex: number,
  inputText: string,

  handleNext: () => void,
  handleInputText: () => void,
};

class Recall extends Component<Props> {

  static mapStateToProps(state) {
    return {
      imageOrder: state.game.imageOrder,
      currIndex: state.recall.currIndex,
      inputText: state.recall.inputText,
    };
  }

  static mapDispatchToProps(dispatch) {
    return {
      handleNext: () => {
        dispatch(next());
      },
      handleInputText: (event) => {
        dispatch(changedInputText(event.target.value));
      },
    };
  }

  render() {
    const { imageData, imageOrder, currIndex, inputText, handleNext, handleInputText } = this.props;
    const numItems = imageOrder.length;

    if (currIndex >= numItems) return null;

    const currCeleb = imageData[imageOrder[currIndex]];

    return (
      <div className="Recall">
        <div className="title">Recall</div>
        <div className="section">
          <img src={currCeleb.image} alt="Celebrity" />
          <br />
          <TextField placeholder="Name" value={inputText} onChange={handleInputText} />
        </div>
        <div className="section">
          <button onClick={handleNext}>Next {`${currIndex + 1} / ${numItems}`}</button>
        </div>
      </div>
    );
  }

}

export default connect(Recall.mapStateToProps, Recall.mapDispatchToProps)(Recall);
