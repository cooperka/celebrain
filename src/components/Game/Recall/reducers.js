import Immutable from 'immutable';

import { actionTypes as gameActionTypes } from '../actions';
import { actionTypes } from './actions';

const initialState = {
  currIndex: 0,

  inputText: '',
  inputs: Immutable.List(),
};

export function recallReducer(state = initialState, { type, payload } = {}) {
  switch (type) {
    case actionTypes.RECALL_NEXT:
      return {
        ...state,
        currIndex: state.currIndex + 1,
        inputText: initialState.inputText,
        inputs: state.inputs.push(state.inputText),
      };

    case actionTypes.CHANGED_INPUT_TEXT:
      return {
        ...state,
        inputText: payload.text,
      };

    case gameActionTypes.RESTART:
      return initialState;

    default:
      return state;
  }
}
