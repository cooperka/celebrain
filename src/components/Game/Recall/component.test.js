// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Recall from './component';

import imageData from '../../../../public/celebs.json';

it('renders', () => {
  const tree = renderer.create(<Recall imageData={imageData} currIndex={0} numItems={1} imageOrder={[0]} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
