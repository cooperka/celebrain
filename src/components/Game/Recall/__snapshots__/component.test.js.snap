// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Recall"
>
  <div
    className="title"
  >
    Recall
  </div>
  <div
    className="section"
  >
    <a
      href="https://commons.wikimedia.org/wiki/File:Nehemiah_Persoff_1960.JPG"
    >
      <img
        alt="Celebrity"
        className="CelebImage-shadowed-1"
        src="https://upload.wikimedia.org/wikipedia/commons/b/ba/Nehemiah_Persoff_1960.JPG"
      />
    </a>
    <br />
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-2"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyPress={undefined}
    >
      <div
        className="MuiInput-root-6 MuiInput-formControl-7 MuiInput-inkbar-8 MuiInput-underline-12"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={true}
          className="MuiInput-input-15"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name={undefined}
          onChange={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Name"
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value={undefined}
        />
      </div>
    </div>
  </div>
  <div
    className="section"
  >
    <button
      className="MuiButtonBase-root-36 MuiButton-root-21"
      disabled={false}
      onBlur={[Function]}
      onClick={undefined}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label-22"
      >
        Next 
        (1/1)
      </span>
      <span
        className="MuiTouchRipple-root-38"
      />
    </button>
  </div>
</div>
`;
